components/core/src/GlobalMySQLMetadataDB.cpp

```diff
@@ -197,10 +197,16 @@ void GlobalMySQLMetadataDB::update_metadata_for_files (const std::string& archiv
     }
 }
 
-GlobalMetadataDB::ArchiveIterator* GlobalMySQLMetadataDB::get_archive_iterator () {
-    auto statement_string = fmt::format("SELECT {} FROM {}{} ORDER BY {} ASC, {} ASC", streaming_archive::cMetadataDB::Archive::Id, m_table_prefix,
-                                        streaming_archive::cMetadataDB::ArchivesTableName, streaming_archive::cMetadataDB::Archive::CreatorId,
-                                        streaming_archive::cMetadataDB::Archive::CreationIx);
+GlobalMetadataDB::ArchiveIterator* GlobalMySQLMetadataDB::get_archive_iterator (epochtime_t begin_ts, epochtime_t end_ts) {
+    auto statement_string = fmt::format("SELECT DISTINCT {}{}.{} FROM {}{} JOIN {}{} ON {}{}.{} = {}{}.{} WHERE {}{}.{} > {} AND {}{}.{} < {} ORDER BY {} ASC, {} ASC",
```

Comment:
A few comments:
1. If I'm not mistaken, I think this condition will miss some relevant archives since it's looking for files (and their corresponding archives) that are completely *within* the time range, rather than *overlapping* with the time range. E.g., if the user is searching for a results in the time range `[5, 10]`, this SQL won't find an archive containing files which have the time range `[4, 6]`. I think ideally we'd make the condition `file_begin_ts <= {user_end_ts} AND file_end_ts >= {user_begin_ts}`.
2. We should be using `>=` and `<=` since the time range from `clg` is inclusive.
3. The DISTINCT and JOIN ops may become expensive when a user has a lot of files. Instead, I think we could compute the begin and end timestamp of an archive at compression time and store it directly in the archives table. I can create a branch demonstrating this if it sounds good to you.

Reply:
Added suggestive insert to line , I think that is what you are expecting ?
Point 3 : Please provide the feature details and branch , we would be happy to look at that.

Can you also add myself and @abvarun226 in contributors to the CLP repo?

Reply:
@kirkrodrigues
1. That makes sense, I've updated the condition in SQL statement to `file_begin_ts <= {user_end_ts} AND file_end_ts >= {user_begin_ts}`
2. I've updated to use inclusive conditions
3. Looks like this will require changes to archives table schema, I'd be happy to incorporate that here. As of current schema, I see files table also has archive_id, just that we will not be able to sort by creation idx

Reply:
1 & 2 look good, thanks! Yeah, 3 requires:
* Adding the begin and end timestamp to the archive schema
* Tracking and updating the begin and end timestamp for the archive in streaming_archive::writer::Archive (probably in `close_segment_and_persist_file_metadata`).
* Changing `GlobalMetadataDB::update_archive_size` to also update the begin and end timestamp.

I'll let you know when I have a branch with those changes so you can take a look or if you feel like you have a rough idea of how to do it, feel free to incorporate it.

Reply:
> Can you also add myself and @abvarun226 in contributors to the CLP repo?

Sorry, forgot to reply to this. Once the PR is merged, GitHub will automatically add anyone who has commits in this PR to the contributors list.

Reply:
Sorry for the wait, here's the commit: https://github.com/kirkrodrigues/clp/commit/4fd12eed6e60613f11d99cf2b0e5bc95cbe87007

If it works for you, I could push it directly to your PR or you can.